api/v1alpha1/storagerequest_types.go

```diff
@@ -24,7 +24,7 @@ import (
 
 // StorageRequestSpec defines the desired state of StorageRequest
 type StorageRequestSpec struct {
-	//+kubebuilder:validation:Enum=blockpool;sharedfilesystem
+	//+kubebuilder:validation:XValidation:rule="self.lowerAscii()=='block'||self.lowerAscii()=='sharedfile'",message="value should be either 'sharedfile' or 'block' (case insensitive)"
```

Comment:
No need to complicate stuff with the mention of case insensitive in the message.  
```suggestion
	//+kubebuilder:validation:XValidation:rule="self.lowerAscii()=='block'||self.lowerAscii()=='sharedfile'",message="value should be either 'sharedfile' or 'block'"
```

Reply:
done.